src/main/java/org/opensearch/knn/index/vectorvalues/KNNVectorValuesIterator.java

```diff
@@ -71,18 +71,28 @@ public interface KNNVectorValuesIterator {
      * {@link DocIdSetIterator} interface. Example: {@link BinaryDocValues}, {@link FloatVectorValues} etc.
      */
     class DocIdsIteratorValues implements KNNVectorValuesIterator {
-        protected DocIdSetIterator docIdSetIterator;
-        private static final List<Function<DocIdSetIterator, Boolean>> VALID_ITERATOR_INSTANCE = List.of(
-            (itr) -> itr instanceof BinaryDocValues,
-            (itr) -> itr instanceof FloatVectorValues,
-            (itr) -> itr instanceof ByteVectorValues
-        );
-
-        DocIdsIteratorValues(@NonNull final DocIdSetIterator docIdSetIterator) {
-            validateIteratorType(docIdSetIterator);
+        private final DocIdSetIterator docIdSetIterator;
+        private KnnVectorValues knnVectorValues = null; // Added reference to KnnVectorValues
+        @Getter
+        @Setter
+        private int lastOrd = -1;
+        @Getter
+        @Setter
+        private Object lastAccessedVector = null;
```

Comment:
same as the old comment, lets try to avoid Object data type. Please have a GH issue on this and should be taken as part of next PR.